host/3.0/buster/amd64/powershell/powershell6-core-tools.Dockerfile

```diff
@@ -0,0 +1,100 @@
+# will be updated to match your local UID/GID (when using the dockerFile property).
+# See https://aka.ms/vscode-remote/containers/non-root-user for details.
+ARG USERNAME=vscode
+ARG USER_UID=1000
+ARG USER_GID=$USER_UID
+
+ARG PS_VERSION=6.2.3
+ARG PS_PACKAGE=powershell_${PS_VERSION}-1.debian.9_amd64.deb
+ARG PS_PACKAGE_URL=https://github.com/PowerShell/PowerShell/releases/download/v${PS_VERSION}/${PS_PACKAGE}
+
+# Download the Linux package and save it
+ADD ${PS_PACKAGE_URL} /tmp/powershell.deb
+
+# Define ENVs for Localization/Globalization
+ENV DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=false \
+    LC_ALL=en_US.UTF-8 \
+    LANG=en_US.UTF-8 \
+    # set a fixed location for the Module analysis cache
+    PSModuleAnalysisCachePath=/var/cache/microsoft/powershell/PSModuleAnalysisCache/ModuleAnalysisCache
+
+ENV FUNCTIONS_WORKER_RUNTIME=powershell
+
+# Configure apt and install packages
+RUN apt-get update \
+    && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get -y install --no-install-recommends apt-utils dialog 2>&1 \
+    && apt install -y /tmp/powershell.deb \
+    #
+    # Verify git and needed tools are installed
+    && apt-get -y install \
+    git \
+    openssh-client \
+    less \
+    iproute2 \
+    procps \
+    curl \
+    apt-transport-https \
+    gnupg2 \
+    lsb-release \
+    # less is required for help in powershell
+    less \
+    # requied to setup the locale
+    locales \
+    # required for SSL
+    ca-certificates \
+    gss-ntlmssp \
+    && apt-get dist-upgrade -y \
+    # enable en_US.UTF-8 locale
+    && sed -i 's/# en_US.UTF-8 UTF-8/en_US.UTF-8 UTF-8/g' /etc/locale.gen \
+    # generate locale
+    && locale-gen && update-locale \
+    # remove powershell package
+    && rm /tmp/powershell.deb \
+    #
+    # Install Azure Functions and Azure CLI
+    && echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $(lsb_release -cs) main" > /etc/apt/sources.list.d/azure-cli.list \
+    && echo "deb [arch=amd64] https://packages.microsoft.com/repos/microsoft-debian-$(lsb_release -cs)-prod $(lsb_release -cs) main" > /etc/apt/sources.list.d/dotnetdev.list \
+    && curl -sL https://packages.microsoft.com/keys/microsoft.asc | (OUT=$(apt-key add - 2>&1) || echo $OUT) \
+    && apt-get update \
+    && apt-get install -y azure-cli azure-functions-core-tools-3 \
+    #
+    # Create a non-root user to use if preferred - see https://aka.ms/vscode-remote/containers/non-root-user.
+    && groupadd --gid $USER_GID $USERNAME \
+    && useradd -s /bin/bash --uid $USER_UID --gid $USER_GID -m $USERNAME \
+    # [Optional] Add sudo support for the non-root user
+    && apt-get install -y sudo \
+    && echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME\
+    && chmod 0440 /etc/sudoers.d/$USERNAME \
+    #
+    # Clean up
+    && apt-get autoremove -y \
+    && apt-get clean -y \
+    && rm -rf /var/lib/apt/lists/* \
+    # intialize powershell module cache
```

Comment:
I think we should do this _after_ installing the Az module.

I'm wondering if there is a good way of forcing cache rebuild. Years ago, the best way was to just invoke a non-existing command, like `IAmSureThisCommandDoesNotExist`: this will make PowerShell look for this command in every available module, so it will rebuild the entire cache and write it to the file within a few seconds. This is a minor optimization, but if we have 10 lines of code just for rebuilding cache anyway, we could as well rebuild it completely.

Reply:
@AnatoliB 
Agree. Clean up should be the last. Very good catch. I'll do that. 
I'm not fully understand invoke the `IAmSureThisCommandDoesNotExist` strategy. So you mean 
L83 - L92 will be 

```bash
&& pwsh -Command "IAmSureThisCommandDoesNotExist" \
```
for forcing powershell cache? I tested in locally, it seems just causing error. Maybe I am wrong. Could you tell me more about the detail?  I also double check the latest way.

Reply:
@TsuyoshiUshio I'm actually talking about performing the `# intialize powershell module cache` part after `# install AzModule`. Clean up can stay where it is, as it should not be affected by installing Az building the cache.

When you invoke a command, PowerShell needs to determine which module contains this command. In order to do that, it locates modules available on the file system, reads and parses module files, etc., until the command is found. This can take long time if you have many large modules installed. In order to mitigate this, PowerShell stores the metadata discovered during this search in a cache, so that next time this data is more readily available. This cache is persisted in a local file. Reading and parsing this local file is much faster than scanning all the modules again.

However, when the command is found, PowerShell stops searching. So, if you have many modules installed, but PowerShell never needed to scan them, they will not be in the cache. The idea behind invoking a non-existing command is to force PowerShell to look at each and every module and build the complete cache, so the first invocation of an Az command will be much faster. This can be achieved by trying to invoke a non-existing command, you just need to ignore the error, something like:
``` PowerShell
try { IAmSureThisCommandDoesNotExist } catch { }
```

Reply:
@AnatoliB 
That's awesome! Great explanation! I understand now. I'll do it! You are the master of PowerShell!